torch/csrc/jit/interpreter.cpp

```diff
@@ -1028,8 +1035,12 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
           } break;
           case TAIL_CALL: {
             af.functions[inst.X]->ensure_defined();
-            const Code &code =
-                af.functions[inst.X]->get_executor().getPlanFor(stack).code;
+            auto num_bailouts = frames.back().function->num_bailouts_ > 0
+                ? frames.back().function->num_bailouts_ - 1
+                : 0;
+            auto& executor = af.functions[inst.X]->get_executor();
+            executor.setNumBailOuts(num_bailouts);
```

Comment:
I feel like it is conceptually simpler for getPlanFor to just take the bailout depth as a parameter. I don't think the executor necessarily needs to know it otherwise.